lessons/getting-started-with-mysql-using-r.md

```diff
@@ -0,0 +1,826 @@
+1. Run a SELECT statement again.  You should not get any rows back.
+2. Re-run the R program above to insert a record.
+3. Perform the SELECT statement.  You should see one row of data.
+
+### Change the INSERT statement to use variables
+
+We will be inserting a lot of data into the table using R, so we will change the INSERT statement to use variables. See the code below the *#Assemble the query* remark.
+
+```
+library(RMySQL)
+
+# R needs a full path to find the settings file
+rmysql.settingsfile<-"C:\\ProgramData\\MySQL\\MySQL Server 5.7\\newspaper_search_results.cnf"
+
+rmysql.db<-"newspaper_search_results"
+storiesDb<-dbConnect(RMySQL::MySQL(),default.file=rmysql.settingsfile,group=rmysql.db) 
+
+# optional - confirms we connected to the database
+dbListTables(storiesDb)
+
+# Assemble the query
+
+# Assign variables
+entryTitle <- "THE LOST LUSITANIA."
+entryPublished <- "21 MAY 1916"
+#convert the string value to a date to store it into the database
+entryPublishedDate <- as.Date(entryPublished, "%d %B %Y")
+entryUrl <- "http://newspapers.library.wales/view/4121281/4121288/94/"
+searchTermsSimple <- "German+Submarine"
+
+# Create the query statement
+query<-paste(
+  "INSERT INTO tbl_newspaper_search_results (
+  story_title,
+  story_date_published,
+  story_url,
+  search_term_used) 
+  VALUES('",entryTitle,"',
+  '",entryPublishedDate,"',
+  LEFT(RTRIM('",entryUrl,"'),99),
+  '",searchTermsSimple,"')",
+  sep = ''
+)
+
+# optional - prints out the query in case you need to troubleshoot it
+print(query)
+
+# execute the query on the storiesDb that we connected to above.
+rsInsert <- dbSendQuery(storiesDb, query)
+
+# disconnect to clean up the connection to the database
+dbDisconnect(storiesDb)
+```
+Let's test this program:
+1. Run a SELECT statement and note the rows you have.
+2. Run the R program above to insert another record.
+3. Perform the SELECT statement.  You should see an additional row of data.
+
+### SQL Errors
+Let's create a simple error in SQL to see what happens.
+
+In R change
+```
+entryTitle <- "THE LOST LUSITANIA."
+```
+to
+```
+entryTitle <- "THE LOST LUSITANIA'S RUDDER."
+```
+and re-run the program.
+
+In the R Console there is an error:
+```
+> rsInsert <- dbSendQuery(storiesDb, query)
+Error in .local(conn, statement, ...) : 
+  could not run statement: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'S RUDDER.',
+  '1916-05-21',
+  LEFT(RTRIM('http://newspapers.library.wales/view/4' at line 6
+```
+You can check with a SELECT statement that there is no record in the table with a story title of THE LOST LUSITANIA'S RUDDER. 
+
+Single apostrophes are part of SQL syntax and they indicate a text value.  If they are in the wrong place, it causes an error.  We have to handle cases where we have data with apostrophes.  SQL accepts two apostrophes in an insert statement to represent an apostrophe in data, as per below:
+```
+''
```

Comment:
Seems overkill to put this as a fenced code block - I might suggest just in-lining it as `''`

Reply:
Fixed.